app/src/main/java/com/keylesspalace/tusky/json/SpannedTypeAdapter.kt

```diff
@@ -32,12 +33,22 @@ import java.lang.reflect.Type
 class SpannedTypeAdapter : JsonDeserializer<Spanned>, JsonSerializer<Spanned?> {
     @Throws(JsonParseException::class)
     override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): Spanned {
-        /* Html.fromHtml returns trailing whitespace if the html ends in a </p> tag, which
-         * all status contents do, so it should be trimmed. */
-        return json.asString?.parseAsHtml()?.trimTrailingWhitespace() ?: SpannedString("")
+        return json.asString
+            /* Mastodon uses 'white-space: pre-wrap;' so spaces are displayed as returned by the Api.
+             * We can't use CSS so we replace spaces with non-breaking-spaces to emulate the behavior.
+             */
+            ?.replace("<br> ", "<br>&nbsp;")
```

Comment:
if those `<br>` are anyway not rendered, why do we need to preserve them?

Reply:
They are rendered. The space after is not. So when we get a status with content like `<p>Test<br> Test</p>` we eat the space after the `<br>` while web displays it.

Reply:
ah, I see!